Tests/Form/Widget/BaseWidgetTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CoreBundle\Tests\Form\Widget;
+
+use Symfony\Bridge\Twig\Extension\FormExtension;
+use Symfony\Bridge\Twig\Extension\TranslationExtension;
+use Symfony\Bridge\Twig\Form\TwigRenderer;
+use Symfony\Bridge\Twig\Form\TwigRendererEngine;
+use Symfony\Bridge\Twig\Tests\Extension\Fixtures\StubFilesystemLoader;
+use Symfony\Bundle\FrameworkBundle\Tests\Templating\Helper\Fixtures\StubTranslator;
+use Symfony\Component\Form\FormExtensionInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\Form\Test\TypeTestCase;
+
+/**
+ * Class BaseWidgetTest.
+ */
+abstract class BaseWidgetTest extends TypeTestCase
+{
+    /**
+     * @var FormExtensionInterface
+     */
+    private $extension;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        $rendererEngine = new TwigRendererEngine(array(
+            'formatter.html.twig',
+            'form_div_layout.html.twig',
+        ));
+
+        if (class_exists('Symfony\Component\Form\Extension\Core\Type\RangeType')) {
+            $csrfManagerClass = 'Symfony\Component\Security\Csrf\CsrfTokenManagerInterface';
+        } else {
+            $csrfManagerClass = 'Symfony\Component\Form\Extension\Csrf\CsrfProvider\CsrfProviderInterface';
+        }
+
+        $renderer = new TwigRenderer($rendererEngine, $this->getMock($csrfManagerClass));
+
+        $this->extension = new FormExtension($renderer);
+
+        //this is ugly workaround for different build strategies and, possibly,
+        //different TwigBridge installation directories
+        $twigPaths = array_filter(array(
+            __DIR__.'/../../../vendor/symfony/twig-bridge/Symfony/Bridge/Twig/Resources/views/Form',
+            __DIR__.'/../../../vendor/symfony/twig-bridge/Resources/views/Form',
+            __DIR__.'/../../../vendor/symfony/symfony/src/Symfony/Bridge/Twig/Resources/views/Form',
+            __DIR__.'/../../../../../symfony/symfony/src/Symfony/Bridge/Twig/Resources/views/Form',
```

Comment:
It would be great to know when we can drop some of these. My guess is there are used when : 

- testing the core bundle, with a non PSR-4 (and therefore old) version of the twig bundle
- testing the core bundle, with a PSR-4 version of the twig bundle
- using the core bundle inside a symfony app (but should running tests be supported in that case ?)
- no idea when this is useful